### PR TITLE
[Commands] Add a helper for writing `init` files.

### PR DESCRIPTION
**Pull request by ddunbar**

### CODE REVIEW

**Comment by ddunbar** [comment]
@swift-ci please test and merge

File: Sources/Commands/init.swift [hunk 1]
```diff
@@ -17,6 +17,16 @@ import func Utility.fputs
 import func Utility.makeDirectories
 import struct Utility.Path
 
+private extension FSProxy {
+    /// Write to a file from a stream producer.
+    mutating func writeFileContents(_ path: String, body: @noescape (OutputByteStream) -> ()) throws {
```
Comment (aciidb0mb3r) [inline Sources/Commands/init.swift]: can this be a generic utility to write files? There is similar need usually

Comment (ddunbar) [inline Sources/Commands/init.swift]: Yup! That is why I factored it out into its own function, but I'm not sure where to put it yet (not sure it belongs as a general purpose extension on FSProxy) so I figured I would leave it here to start.

Where do you think it should go?

**Comment by ddunbar** [comment]
Tests passed, this just hit the Foundation JSON decoding bug.

File: Sources/Commands/init.swift (hunk 1 above)
Comment (aciidb0mb3r) [inline Sources/Commands/init.swift]: I think it can go in utility or you want to get rid of entire utility module?

Comment (ddunbar) [inline Sources/Commands/init.swift]: Yeah I am trying to gradually kill Utility by making the APIs available on more structured types and with review.

**Pull request merged**